### ISSUE TITLE
add self-hosted plan for self-hosted flavor

### ISSUE THREAD

**Issue by darkskygit** [toeverything/AFFiNE#6156]

**Comment by greedyhao** [comment]
Will self-host plan be a one-time fee?

**Comment by pengx17** [comment]
> Will self-host plan be a one-time fee?

@greedyhao I am not sure what is your question. Do you mean if there is a license charge fee for self-hosting?

**Comment by EYHN** [comment]
> Will self-host plan be a one-time fee?

Of course, it's free.

The self-host plan have unlimited storage space for free.

We may impose some limit on the number of collaborators to prevent enterprises from directly using the self-host version, as we hope to generate some revenue from business users. There will be no limit for personal use.

**Comment by greedyhao** [comment]
> > Will self-host plan be a one-time fee?
> 
> Of course, it's free.
> 
> The self-host plan have unlimited storage space for free.
> 
> We may impose some limit on the number of collaborators to prevent enterprises from directly using the self-host version, as we hope to generate some revenue from business users. There will be no limit for personal use.

Thanks a lot. That's what I want to know.

**Comment by gymnae** [comment]
> > Will self-host plan be a one-time fee?
> 
> Of course, it's free.
> 
> The self-host plan have unlimited storage space for free.
> 
> We may impose some limit on the number of collaborators to prevent enterprises from directly using the self-host version, as we hope to generate some revenue from business users. There will be no limit for personal use.

This reads a bit contradictory to the comment made in the related discussion. Are you saying that hobbyists, non-commercial users will have no limitation on collaboratiors per space and globally and storage?

**Comment by darkskygit** [comment]
> > > Will self-host plan be a one-time fee?
> > 
> > 
> > Of course, it's free.
> > The self-host plan have unlimited storage space for free.
> > We may impose some limit on the number of collaborators to prevent enterprises from directly using the self-host version, as we hope to generate some revenue from business users. There will be no limit for personal use.
> 
> This reads a bit contradictory to the comment made in the related discussion. Are you saying that hobbyists, non-commercial users will have no limitation on collaboratiors per space and globally and storage?

collaborators always have seat limits

**Comment by gymnae** [comment]
Ok, let me rephrase it: Will the space owner on a self-hosted install without a commercial usage be restricted in the number of people they can invite to participate in a space? Currently that limit is three.

**Comment by darkskygit** [comment]
> Ok, let me rephrase it: Will the space owner on a self-hosted install without a commercial usage be restricted in the number of people they can invite to participate in a space? Currently that limit is three.

tldr, will. however, it needs to be emphasized that "seats" are currently equivalent to "people who can access the whole workspace" and "people who can collaborate together in the workspace", but in the future, we will support sharing a single doc and allow you to edit and collaborate it with others that not in seats. in this case, the collaborators of single doc will not be limited by the number of seats.

this feature is not yet implemented, but we will support it in a future version.

**Comment by hilsonp** [comment]
> > Ok, let me rephrase it: Will the space owner on a self-hosted install without a commercial usage be restricted in the number of people they can invite to participate in a space? Currently that limit is three.
> 
> tldr, will. however, it needs to be emphasized that "seats" are currently equivalent to "people who can access the whole workspace" and "people who can collaborate together in the workspace", but in the future, we will support sharing a single doc and allow you to edit and collaborate it with others that not in seats. in this case, the collaborators of single doc will not be limited by the number of seats.
> 
> this feature is not yet implemented, but we will support it in a future version.

Hello !
Thank you very much for the self-hosted version. This is much appreciated.
I'm not sure I fully understand the comment.
I'm ready to install Affine on my homelab.
This is aimed as being our family wiki where we can all access and edit information.
Will that be possible or should I disown 2 of my children 😅 ?

**Comment by darkskygit** [comment]
> Hello ! Thank you very much for the self-hosted version. This is much appreciated. I'm not sure I fully understand the comment. I'm ready to install Affine on my homelab. This is aimed as being our family wiki where we can all access and edit information. Will that be possible or should I disown 2 of my children 😅 ?

we will support purchasing seats in a future version, so stay tuned

**Comment by gymnae** [comment]
And purchasing will be required no matter the usage of a self-hosted install? aka private/hobbyist/non-commercial usage?
Do you have a cost estimate and will it be a one time fee or subscriprion?

**Comment by darkskygit** [comment]
> And purchasing will be required no matter the usage of a self-hosted install? aka private/hobbyist/non-commercial usage? Do you have a cost estimate and will it be a one time fee or subscriprion?

once the relevant details are confirmed, we will release it on our official media account and blog, please stay tuned

**Comment by hilsonp** [comment]
For example, Apple allows iCloud to be shared between 6 family members 🙏 (for private use only).

**Comment by gymnae** [comment]
Outline has a good pricing concept for a comparable product: https://www.getoutline.com/pricing

**Comment by darkskygit** [comment]
> For example, Apple allows iCloud to be shared between 6 family members 🙏 (for private use only).

the current pro plan of AFFiNE Cloud supports collaboration with 10 people

**Comment by darkskygit** [comment]
> Outline has a good pricing concept for a comparable product: https://www.getoutline.com/pricing

thank you for your suggestion, we will refer to the relevant information

**Comment by nansenat16** [comment]
I was planning to move from outline (outline free plan cannot create workspace) to affine, but after installation, I found that the license plan is connected to the cloud. I gave up and returned to notion. Notion’s 8 USD plan can add 100 guests, if it is not unlimited of users (EX: <500) affine self-hosting is not competitive

**Comment by darkskygit** [comment]
> I was planning to move from outline (outline free plan cannot create workspace) to affine, but after installation, I found that the license plan is connected to the cloud. I gave up and returned to notion. Notion’s 8 USD plan can add 100 guests, if it is not unlimited of users (EX: <500) affine self-hosting is not competitive

<img width="1097" alt="image" src="https://github.com/toeverything/AFFiNE/assets/25152247/5b3b122c-2ef1-4e82-9ebe-9b2439eff0b9">

According to the definition of Notion, the user who can edit the entire Workspace is Member. Guest specifically refers to a user who can edit a certain page.

At present, AFFiNE does not have a single-page collaboration feature for the time being, so it cannot be compared with this number.

The single-page collaboration feature will be launched later this year. Please follow our social media accounts and blogs.

**Comment by almereyda** [comment]
> Outline has a good pricing concept for a comparable product: https://www.getoutline.com/pricing

It is to second that the license used for Outline (BSL-1.1) with the terms supplied (1) allows unlimited usage for personal and not-for-profit use and (2) releases the source code under a free license after two years.

- https://github.com/outline/outline/blob/ceb7ae15142bba38b7107b7a19820dbb1d200300/LICENSE

This is a model that works very well to support the civic society, while allowing to constrain the usage on commercial vendors offering services based on the product. The dual licensing of freely releasing the client but enclosing the server to proprietary terms (https://github.com/toeverything/AFFiNE/blob/canary/packages/backend/server/LICENSE) breaks the collaboration model of free software, in which error reports and triaging are contributed back to upstream, *because* we know it benefits others.

I would like to suggest to consider fine-tuning and adopting the current licensing approach with regards to personal and not-for-profit use cases that benefit society as a whole, but not only the sole proprietor of the product.

**Comment by gymnae** [comment]
Since there's unclear communication and diverging information regarding licensing, I will go back to Nextcloud Collectives and outline for the moment. Sad, because this looks promising, but claiming it's fully open source* when it isn't and free for self-hosting when it isn't** is a bit of a deal-breaker.

Plus: All data is stored in a database in a custom format, Markdown is only possible via export. So the data cannot be transported that easily and can get lost, if the database crashes (looking at you, WordPress)

* in parts
** with strong limitations, which are not yet well-defined, but appear to become quite strict

**Comment by darkskygit** [comment]
the original intention of this issue is to "promote the repair of the self-hosted plan" rather than "what feature should self-hosted have?" or "what license should an open source project have?"
there are already too many off-topic discussions, i am here Mark all off-topic discussions as `off-topic`.

if you have any questions and discussions related to self-hosted, please go to: https://github.com/toeverything/AFFiNE/discussions to discuss

**Comment by alissa-tung** [comment]
Hello, curious about is there progress on it? Is self host version can still only have three users in one workspace, and the space is limited to 10 GiB?

**Comment by Artaud** [comment]
> Hello, curious about is there progress on it? Is self host version can still only have three users in one workspace, and the space is limited to 10 GiB?

You can set your users to be on the Pro plan manually in the database, which raises the limits.

**Comment by alissa-tung** [comment]
> > Hello, curious about is there progress on it? Is self host version can still only have three users in one workspace, and the space is limited to 10 GiB?
> 
> You can set your users to be on the Pro plan manually in the database, which raises the limits.

Thank you, I had connect to the self-host pg, would it be convenient to share which table has the field for pro? thank you

**Comment by Artaud** [comment]
It's not hard to figure out - the `user_features` table has a `feature_id` column which links the user to a specific plan, listed in the `features` table. 
I have my users set to 6, which is the `pro_plan_v1`.

**Comment by alissa-tung** [comment]
> It's not hard to figure out - the `user_features` table has a `feature_id` column which links the user to a specific plan, listed in the `features` table. I have my users set to 6, which is the `pro_plan_v1`.

Thank you, save my day!

**Comment by alissa-tung** [comment]
from https://github.com/toeverything/AFFiNE/blob/a14194c482fa25f2443cdcb5e180c41f43ad7a84/packages/backend/server/src/core/quota/schema.ts#L37

~~seems that even with `pro_plan_v1`, `memberLimit` is set to 10, maybe I should package an image my self~~

NO need for a new docker image

---

The correct `feature_id` for `user_features` is in table `features`, which may look like

```sh
docker exec -it affine_postgres psql -U affine
```

```
affine=# select id, feature, configs from features where feature = 'pro_plan_v1';
 id |   feature   |                                                    configs                                                    
----+-------------+---------------------------------------------------------------------------------------------------------------
  6 | pro_plan_v1 | {"name":"Pro","blobLimit":104857600,"storageQuota":107374182400,"historyPeriod":2592000000,"memberLimit":100}
 11 | pro_plan_v1 | {"name":"Pro","blobLimit":104857600,"storageQuota":107374182400,"historyPeriod":2592000000,"memberLimit":100}
(2 rows)
```

in the `configs`, update JSON to one with larger `memberLimit` is enough.

after that, reboot the docker compose
